arxiv/license.py

```diff
@@ -0,0 +1,114 @@
+
+    'http://creativecommons.org/licenses/by-nc-sa/4.0/': {
+        'uri': 'http://creativecommons.org/licenses/by-nc-sa/4.0/',
+        'order': 7,
+        'label': 'Creative Commons Attribution-Noncommercial-ShareAlike '
+        ' license',
+        'is_current': True,
+        'is_valid': True,
+        'icon_uri': f'{LICENSE_ICON_BASE_URI}/by-nc-sa-4.0.png'},
+
+    'http://creativecommons.org/publicdomain/zero/1.0/': {
+        'uri': 'http://creativecommons.org/publicdomain/zero/1.0/',
+        'label': 'Creative Commons Public Domain Declaration',
+        'order': 8,
+        'is_current': True,
+        'is_valid': True,
+        'icon_uri': f'{LICENSE_ICON_BASE_URI}/zero-1.0.png'
+    },
+
+    'http://arxiv.org/licenses/assumed-1991-2003/': {
+        'uri': 'http://arxiv.org/licenses/assumed-1991-2003/',
+        'label': 'Assumed arXiv.org perpetual, non-exclusive license to '
+                 'distribute this article for submissions made before '
+                 'January 2004',
+        'order': 9,
+        'is_current': False,
+        'is_valid': True,
+    },
+
+    'http://creativecommons.org/licenses/by/3.0/': {
+        'uri': 'http://creativecommons.org/licenses/by/3.0/',
+        'label': 'Creative Commons Attribution license',
+        'order': 2,
+        'is_current': False,
+        'is_valid': True,
+        'icon_uri': f'{LICENSE_ICON_BASE_URI}/by-3.0.png'
+    },
+
+    'http://creativecommons.org/licenses/by-nc-sa/3.0/': {
+        'uri': 'http://creativecommons.org/licenses/by-nc-sa/3.0/',
+        'label': 'Creative Commons Attribution-Noncommercial-ShareAlike '
+        'license',
+        'order': 3,
+        'is_current': False,
+        'is_valid': True,
+        'icon_uri': f'{LICENSE_ICON_BASE_URI}/by-nc-sa-3.0.png'
+    },
+    'http://creativecommons.org/licenses/publicdomain/': {
+        'uri': 'http://creativecommons.org/licenses/publicdomain/',
+        'label': 'Creative Commons Public Domain Declaration',
+        'note': '(Suitable for US government employees, for example)',
+        'order': 4,
+        'is_current': False,
+        'is_valid': True,
+        'icon_uri': "$icon_baseurl/publicdomain.png"
+    },
+    'no_license': {
+        'order': 99,
+        'label': 'I do not certify that any of the above licenses apply',
+        'is_valid': False
+    }
+}
+
+ASSUMED_LICENSE = LICENSES['http://arxiv.org/licenses/assumed-1991-2003/']
+
+# Historical license to updated/current license (old: new)
+TRANSLATED_LICENSES = {
+    'http://creativecommons.org/licenses/by/3.0/':
+    'http://creativecommons.org/licenses/by/4.0/',
+    'http://creativecommons.org/licenses/by-nc-sa/3.0/':
+    'http://creativecommons.org/licenses/by-nc-sa/4.0/',
+    'http://creativecommons.org/licenses/publicdomain/':
+    'http://creativecommons.org/publicdomain/zero/1.0/'
+}
+
+CURRENT_LICENSES = {
+    k: v for k, v in LICENSES.items()
+    if all(k in v for k in ('is_current', 'order')) and v['is_current']
```

Comment:
This is totally nit-picky, but any reason not to make this more brutish for the sake of clarity? eg.

```if 'is_current' in v and 'order' in v and v['is_current']```

Reply:
Agree!